universalmutator/genmutants.py

```diff
@@ -361,26 +391,33 @@ def main():
         if (lineFile is not None) and mutant[0] not in lines:
             # skip if not a line to mutate
             continue
-        if (not noFastCheck) and (mutant[0] not in interestingLines) and (mutant[0] not in deadCodeLines):
+        if (not noFastCheck) and (not comby) and (mutant[0] not in interestingLines) and (mutant[0] not in deadCodeLines):
             fastCheckMutant = (mutant[0], toGarbage(source[mutant[0] - 1]))
             mutator.makeMutant(source, fastCheckMutant, tmpMutantName)
-            if compileFile is None:
-                mutantResult = handler(tmpMutantName, mutant, sourceFile, uniqueMutants)
-            else:
-                mutantResult = handler(tmpMutantName, mutant, sourceFile, uniqueMutants, compileFile=compileFile)
-            if mutantResult in ["VALID", "REDUNDANT"]:
-                deadCodeLines.append(mutant[0])
-                print("LINE", str(mutant[0]) + ":", source[mutant[0] - 1][:-1], end=" ")
-                print("APPEARS TO BE COMMENT OR DEAD CODE, SKIPPING...")
-            else:
-                interestingLines.append(mutant[0])
-        if mutant[0] in deadCodeLines:
+            fastCheckLine(mutant, source, sourceFile, uniqueMutants, compileFile, handler, deadCodeLines, interestingLines, tmpMutantName, mutant[0])
+        if (not noFastCheck) and comby:
+            checkLines = []
+            for i in range(mutant[3][0], mutant[3][1] + 1):
+                if i not in deadCodeLines or i not in interestingLines:
+                    checkLines.append(i)
+            for lineNo in checkLines:
+                fastCheckMutant = (lineNo, toGarbage(source[lineNo - 1]))
+                mutator.makeMutant(source, fastCheckMutant, tmpMutantName)
+                fastCheckLine(mutant, source, sourceFile, uniqueMutants, compileFile, handler, deadCodeLines, interestingLines, tmpMutantName, lineNo)
+        if (not noFastCheck) and (not comby) and mutant[0] in deadCodeLines:
+            continue
+        if (not noFastCheck) and comby and checkCombyDeadCode(deadCodeLines, mutant):
             continue
-        print("PROCESSING MUTANT:",
+        if comby:
+            sourceJoined = ''.join(source)
+            print("PROCESSING MUTANT:",
+              "range" + str(mutant[0]) + ":", sourceJoined[mutant[0][0]:mutant[0][1]].replace("\n", "\\n"), " ==> ", mutant[1], end="...")
+        else:
+            print("PROCESSING MUTANT:",
               str(mutant[0]) + ":", source[mutant[0] - 1][:-1], " ==> ", mutant[1][:-1], end="...")
-        if showRules:
+        if (not comby) and showRules:
             print("(FROM:", mutant[2][1], end=")...")
-        mutator.makeMutant(source, mutant, tmpMutantName)
+        mutator.makeMutantComby(sourceJoined, mutant, tmpMutantName) if comby else mutator.makeMutant(source, mutant, tmpMutantName)
```

Comment:
nit: I'd just make this into a standard `if/else` block. I find reading the inlined code to be a bit of a speed bump when reading

Reply:
Updating this to not used inline code.